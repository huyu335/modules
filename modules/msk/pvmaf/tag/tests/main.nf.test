// TODO nf-core: Once you have added the required tests, please run the following command to build this file:
// nf-core modules test pvmaf/tag
nextflow_process {

    name "Test Process PVMAF_TAG"
    script "../main.nf"
    process "PVMAF_TAG"
    config "./nextflow.config"

    tag "modules"
    tag "modules_nfcore"
    tag "modules_msk"
    tag "pvmaf"
    tag "pvmaf/tag"

    // TODO nf-core: Change the test name preferably indicating the test-data and file-format used
    test("chr22maf - msk") {

        // TODO nf-core: If you are created a test for a chained module
        // (the module requires running more than one process to generate the required output)
        // add the 'setup' method here.
        // You can find more information about how to use a 'setup' method in the docs (https://nf-co.re/docs/contributing/modules#steps-for-creating-nf-test-for-chained-modules).

        when {
            process {
                """
                
                input[0] = [
                            [ id:'chr22', patient:'test'], // meta map
                            file(params.test_data_msk['chr22']['chr22maf'], checkIfExists: true),
                        ]
                input[1] = 'traceback'
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
                //TODO nf-core: Add all required assertions to verify the test output.
                // See https://nf-co.re/docs/contributing/tutorials/nf-test_assertions for more information and examples.
            )
        }

    }

    // TODO nf-core: Change the test name preferably indicating the test-data and file-format used but keep the " - stub" suffix.
    test("chr22maf - msk - stub") {

        options "-stub"

        when {
            process {
                """
                
                input[0] = [
                            [ id:'chr22', patient:'test'], // meta map
                            file(params.test_data_msk['chr22']['chr22maf'], checkIfExists: true),
                        ]
                input[1] = 'traceback'
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
                //TODO nf-core: Add all required assertions to verify the test output.
            )
        }

    }

}
