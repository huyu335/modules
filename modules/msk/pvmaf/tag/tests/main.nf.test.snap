{
    "chr22maf - msk - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "chr22",
                            "patient": "test"
                        },
                        "test_traceback.maf:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,f8facbd0f81020509538710bac6b9f8e"
                ],
                "maf": [
                    [
                        {
                            "id": "chr22",
                            "patient": "test"
                        },
                        "test_traceback.maf:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,f8facbd0f81020509538710bac6b9f8e"
                ]
            }
        ],
        "timestamp": "2024-02-22T13:39:51.876899"
    },
    "chr22maf - msk": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "chr22",
                            "patient": "test"
                        },
                        "test_traceback.maf:md5,23ca5bc6918acfd84b98fdce78bc008b"
                    ]
                ],
                "1": [
                    "versions.yml:md5,f8facbd0f81020509538710bac6b9f8e"
                ],
                "maf": [
                    [
                        {
                            "id": "chr22",
                            "patient": "test"
                        },
                        "test_traceback.maf:md5,23ca5bc6918acfd84b98fdce78bc008b"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,f8facbd0f81020509538710bac6b9f8e"
                ]
            }
        ],
        "timestamp": "2024-02-22T14:32:56.63193"
    }
}